ZelBack/src/services/appSystem/systemIntegration.js

```diff
@@ -0,0 +1,487 @@
+const os = require('os');
+const config = require('config');
+const log = require('../../lib/log');
+const messageHelper = require('../messageHelper');
+const serviceHelper = require('../serviceHelper');
+const verificationHelper = require('../verificationHelper');
+const dockerService = require('../dockerService');
+const daemonServiceFluxnodeRpcs = require('../daemonService/daemonServiceFluxnodeRpcs');
+const fluxNetworkHelper = require('../fluxNetworkHelper');
+const benchmarkService = require('../benchmarkService');
+const hwRequirements = require('../appRequirements/hwRequirements');
+const daemonServiceBenchmarkRpcs = require('../daemonService/daemonServiceBenchmarkRpcs');
+const generalService = require('../generalService');
+
+// Node specifications cache
+let nodeSpecs = {
+  cpuCores: 0,
+  ram: 0,
+  ssdStorage: 0,
+};
+
+/**
+ * Get node specifications (CPU, RAM, Storage) and cache them
+ * @returns {Promise<void>}
+ */
+async function getNodeSpecs() {
+  try {
+    if (nodeSpecs.cpuCores === 0) {
+      nodeSpecs.cpuCores = os.cpus().length;
+    }
+    if (nodeSpecs.ram === 0) {
+      nodeSpecs.ram = os.totalmem() / 1024 / 1024; // Convert to MB
+    }
+    if (nodeSpecs.ssdStorage === 0) {
+      // get my external IP and check that it is longer than 5 in length.
+      const benchmarkResponse = await daemonServiceBenchmarkRpcs.getBenchmarks();
+      if (benchmarkResponse.status === 'success') {
+        const benchmarkResponseData = JSON.parse(benchmarkResponse.data);
+        log.info(`Gathered ssdstorage ${benchmarkResponseData.ssd}`);
+        nodeSpecs.ssdStorage = benchmarkResponseData.ssd;
```

Comment:
function says it should get node specifications (CPU, RAM, Storage), also comment says we should get `external IP`, but it only gets ssdStorage.